src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private static final int MAX_FIGURE_SIZE = 20;
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+
+        int figureType = random.nextInt(NUMBER_OF_FIGURES);
+
+        switch (figureType) {
+            case 0:
+                figure = new Circle(colorSupplier.getRandomColor().toLowerCase(),
+                        random.nextInt(MAX_FIGURE_SIZE) + 1);
+                return figure;
```

Comment:
```suggestion
                return new Circle(colorSupplier.getRandomColor().toLowerCase(),
                        random.nextInt(MAX_FIGURE_SIZE) + 1);
```
і так ми можемо видалити поле 'Figure figure;'